test/test_compiler.txt

```diff
@@ -11508,3 +11508,146 @@ ontimer(date=[set_time(new Date(2021, 0, 1), new Time(10, 30, 0))]) => @org.thin
   } finally {
     await __env.exitProcedure(0, null);
   }
+====
+// 114
+@com.spotify2.song() join @com.spotify2.album() on first.album == second.id;
+>>>>
+  "use strict";
+  let _t_0;
+  let _t_1;
+  let _t_2;
+  let _t_3;
+  let _t_4;
+  let _t_5;
+  let _t_6;
+  let _t_7;
+  let _t_8;
+  let _t_9;
+  let _t_10;
+  let _t_11;
+  let _t_12;
+  let _t_13;
+  let _t_14;
+  let _t_15;
+  let _t_16;
+  let _t_17;
+  let _t_18;
+  let _t_19;
+  let _t_20;
+  let _t_21;
+  let _t_22;
+  let _t_23;
+  let _t_24;
+  let _t_25;
+  let _t_26;
+  let _t_27;
+  let _t_28;
+  let _t_29;
+  let _t_30;
+  let _t_31;
+  let _t_32;
+  let _t_33;
+  let _t_34;
+  let _t_35;
+  let _t_36;
+  let _t_37;
+  let _t_38;
+  let _t_39;
+  let _t_40;
+  let _t_41;
+  let _t_42;
+  let _t_43;
+  let _t_44;
+  let _t_45;
+  let _t_46;
+  await __env.enterProcedure(0, null);
+  try {
+    _t_0 = async function(__emit) {
+      try {
+        _t_1 = {};
+        _t_2 = await __env.invokeQuery("com.spotify2", { }, "song", _t_1, { projection: [] });
+        _t_3 = __builtin.getAsyncIterator(_t_2);
+        {
+          let _iter_tmp = await _t_3.next();
+          while (!_iter_tmp.done) {
+            _t_4 = _iter_tmp.value;
+            _t_5 = _t_4[0];
+            _t_6 = _t_4[1];
+            _t_7 = _t_6.__response;
+            _t_8 = _t_6.id;
+            _t_9 = _t_6.artists;
+            _t_10 = _t_6.album;
+            _t_11 = _t_6.genres;
+            _t_12 = _t_6.release_date;
+            _t_13 = _t_6.popularity;
+            _t_14 = _t_6.energy;
+            _t_15 = _t_6.danceability;
+            __emit(_t_5, _t_6);
+            _iter_tmp = await _t_3.next();
+          }
+        }
+      } catch(_exc_) {
+        __env.reportError("Failed to invoke query", _exc_);
+      }
+    }
+    _t_16 = async function(__emit) {
+      try {
+        _t_17 = {};
+        _t_18 = await __env.invokeQuery("com.spotify2", { }, "album", _t_17, { projection: [] });
+        _t_19 = __builtin.getAsyncIterator(_t_18);
+        {
+          let _iter_tmp = await _t_19.next();
+          while (!_iter_tmp.done) {
+            _t_20 = _iter_tmp.value;
+            _t_21 = _t_20[0];
+            _t_22 = _t_20[1];
+            _t_23 = _t_22.__response;
+            _t_24 = _t_22.id;
+            _t_25 = _t_22.artists;
+            _t_26 = _t_22.release_date;
+            _t_27 = _t_22.popularity;
+            __emit(_t_21, _t_22);
+            _iter_tmp = await _t_19.next();
+          }
+        }
+      } catch(_exc_) {
+        __env.reportError("Failed to invoke query", _exc_);
+      }
+    }
+    _t_28 = __builtin.tableJoin(_t_0, _t_16);
+    {
+      let _iter_tmp = await _t_28.next();
+      while (!_iter_tmp.done) {
+        _t_29 = _iter_tmp.value;
+        _t_30 = _t_29[0];
+        _t_31 = _t_29[1];
+        _t_32 = _t_31.first.__response;
+        _t_33 = _t_31.first.id;
```

Comment:
Wait no!
This needs to be `_t_31["first.id"]`, because there is no "first" object, it's a flat set of prefixed properties.
Which, btw, will cause issues with the dialogue result syntax I suspect, because you have identifiers with dots in them. Can you add a test for that one too?